lib/metric_fu/reporting/result.rb

```diff
@@ -11,14 +11,20 @@ def self.result
   #
   # It tracks the results generated by each metric used in this test run.
   class Result
+    if  RUBY_VERSION == "1.9.2"
+      # Psych is required for YAML.to_json to work
+      # In 1.9.2, the default is still Syck
+      YAML::ENGINE.yamler = "psych"
+    end
+
     # Renders the result of the result_hash into a yaml serialization
     # ready for writing out to a file.
     #
     # @return YAML
     #   A YAML object containing the results of the result generation
     #   process
     def as_yaml
-      result_hash.to_yaml
+      YAML.to_json(result_hash)
```

Comment:
@bf4 This seems a bit confusing, shouldn't we try to rename the method to be `as_json` instead?